Documentation on performance enhancement changes (#2068)
forcedotcom/salesforcedx-vscode: docs/_articles/en/user-guide/perf-enhancements.md

---
title: Performance enhancements
lang: en
---

> The performance enhancement changes allows you to run single file metadata deploys through a more efficient code path. These changes currently only support Apex Class metadata types. More metadata types will get included in the following releases.

Reviewer: ```suggestion
The performance enhancement changes allow you to run single file metadata deploys through a more efficient code path. These changes currently only support Apex Class metadata types. Additional metadata types will be supported in future releases.
```

> NOTICE: Performance enhancement changes for single Apex Class file deploys is currently in beta. If you find any bugs or have feedback, [open a GitHub issue](./en/bugs-and-feedback).

## Setup

> Because the performance enhancement changes are in beta, you must enable them:

Reviewer: ```suggestion
To enable this beta feature:
```

1. Select **File** > **Preferences** > **Settings** (Windows or Linux) or **Code** > **Preferences** > **Settings** (macOS).
1. Under Salesforce Feature Previews, select Experimental: Deploy Retrieve.

> In this beta release, performance enhancements will only occur when running the **SFDX: Deploy this Source to Org** command from an Apex Class.

Reviewer 2: Should we specify which code paths are affected? ie. the right click menus and the command palette invokations?

Author: In both code paths the command name is `SFDX: Deploy this Source to Org`, I think we can add it or leave it as is. Whatever you think makes it clearer.

> In this beta release, we have released performance enhancement changes when running the **SFDX: Deploy this Source to Org** command only when running it for Apex Class.

Reviewer: ```suggestion
In this beta release, performance enhancements will only occur when running the **SFDX: Deploy this Source to Org**  command from an Apex Class.
```

Reviewer 3: How about? "In this beta release, the performance enhancements are only enabled for **SFDX: Deploy this Source to Org**  command from an Apex Class."
